feature_engine/selection/probe_feature_selection.py

```diff
@@ -0,0 +1,277 @@
+    calculates the average feature importance of all the probe features.
+
+    The transformer ranks the variables based on their feature importances. The
+    variables that have a feature importance less than the feature importance or
+    average feature importance of the probe feature(s) are dropped from the dataset.
+
+    Parameters
+    ----------
+    {estimator}
+
+    {variables}
+
+    {scoring}
+
+    n_probes: int, default=1
+        Number of probe features to be created. If distribution is 'all',
+        n_probes must be a multiple of 3.
+
+    distribution: str, default='normal'
+        The distribution used to create the probe features. The options are
+        'normal', 'binomial', 'uniform', and 'all'. 'all' creates at least 1 or more
+        probe features comprised of each distribution type, i.e., normal, binomial,
+        and uniform. The remaining options create `n_probes` features of the selected
+        distribution
+
+    {cv}
+
+    Attributes
+    ----------
+    probe_features_:
+        A dataframe comprised of the pseudo-randomly generated features based
+        on the selected distribution.
+
+    {feature_importances_}
+
+    {features_to_drop_}
+
+    {variables_}
+
+    {feature_names_in_}
+
+    {n_features_in_}
+
+    Methods
+    -------
+    {fit}
+
+    {fit_transform}
+
+    {get_support}
+
+    {transform}
+    """
+
+    def __init__(
+        self,
+        estimator,
+        variables: Variables = None,
+        scoring: str = "roc_auc",
+        n_probes: int = 1,
+        distribution: str = "normal",
+        cv: int = 5,
+        random_state: int = 0,
+        confirm_variables: bool = False,
+    ):
+        if distribution not in ["normal", "binary", "uniform", "all"]:
+            raise ValueError(
+                "distribution takes on 'normal', 'binary', 'uniform', or 'all' as "
+                f"values. Got {distribution} instead."
+            )
+
+        if distribution == "all" and n_probes % 3 != 0:
+            raise ValueError(
+                "If distribution is 'all' the n_probes must be a multiple of 3. "
+                f"Got {n_probes} instead."
+            )
+
+        if not isinstance(n_probes, int):
+            raise ValueError(f"n_probes must be an integer. Got {n_probes} instead.")
+
+        super().__init__(confirm_variables)
+        self.estimator = estimator
+        self.variables = variables
+        self.scoring = scoring
+        self.distribution = distribution
+        self.cv = cv
+        self.n_probes = n_probes
+        self.random_state = random_state
+
+    def fit(self, X: pd.DataFrame, y: pd.Series):
+        """
+        Find the important features.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+
+        y: array-like of shape (n_samples)
+                   Target variable. Required to train the estimator.
```

Comment:
I think this has extra indentation and may cause problems when rebuilding the docs.